mmeval/metrics/coco_pose.py

```diff
@@ -0,0 +1,627 @@
+
+    Examples:
+        >>> import numpy as np
+        >>> import copy
+        >>> from mmeval.fileio import load
+        >>> from mmeval import CocoPoseMetric
+        >>> try:
+        ...     from xtcocotools.coco import COCO
+        ...     from xtcocotools.cocoeval import COCOeval
+        ...     HAS_XTCOCOTOOLS = True
+        ... except ImportError:
+        ...     HAS_XTCOCOTOOLS = False
+        ...
+        >>> ann_file = 'tests/test_metrics/data/coco_pose_sample.json'
+        >>> coco = COCO(ann_file)
+        loading annotations into memory...
+        Done (t=0.00s)
+        creating index...
+        index created!
+        >>> classes = coco.loadCats(coco.getCatIds())
+        >>> sigmas = np.array([
+        ...     0.026, 0.025, 0.025, 0.035, 0.035, 0.079, 0.079, 0.072, 0.072,
+        ...     0.062, 0.062, 0.107, 0.107, 0.087, 0.087, 0.089, 0.089
+        ... ]).astype(np.float32)
+        >>> coco_dataset_meta = {
+        ...     'CLASSES': classes,
+        ...     'num_keypoints': 17,
+        ...     'sigmas': sigmas,
+        ... }
+        >>> def _convert_ann_to_pred_and_gt(ann_file):
+        ...     predictions = []
+        ...     groundtruths = []
+        ...     db = load(ann_file)
+        ...     imgid2info = dict()
+        ...     for img in db['images']:
+        ...         imgid2info[img['id']] = img
+        ...     for ann in db['annotations']:
+        ...         bboxes = np.array(ann['bbox'], dtype=np.float32).reshape(-1, 4)
+        ...         keypoints = np.array(ann['keypoints']).reshape((1, -1, 3))
+        ...         prediction = {
+        ...             'id': ann['id'],
+        ...             'img_id': ann['image_id'],
+        ...             'bboxes': bboxes,
+        ...             'keypoints': keypoints[..., :2],
+        ...             'keypoint_scores': keypoints[..., -1],
+        ...             'bbox_scores': np.ones((1, ), dtype=np.float32),
+        ...         }
+        ...         groundtruth = {
+        ...             'img_id': ann['image_id'],
+        ...             'width': 640,
+        ...             'height': 480,
+        ...             'num_keypoints': ann['num_keypoints'],
+        ...             'raw_ann_info': [copy.deepcopy(ann)],
+        ...         }
+        ...         if 'area' in ann:
+        ...             groundtruth['area'] = ann['area']
+        ...         if 'crowdIndex' in imgid2info[ann['image_id']]:
+        ...             groundtruth['crowd_index'] = imgid2info[
+        ...                 ann['image_id']]['crowdIndex']
+        ...         predictions.append(prediction)
+        ...         groundtruths.append(groundtruth)
+        ...     return predictions, groundtruths
+        >>> predictions, groundtruths = _convert_ann_to_pred_and_gt(ann_file)
+        >>> coco_pose_metric = CocoPoseMetric(
+        ...     ann_file=ann_file,
+        ...     dataset_meta=coco_dataset_meta)
```

Comment:
If `dataset_meta` is required in this metric, I think it should be explained in docstring